trestle/transforms/implementations/osco.py

```diff
@@ -577,66 +577,107 @@ def __init__(
-            value = parameter_value.__root__
-            rationale = self._get_rationale_for_set_value()
-            set_value = {'name': name, 'value': value, 'rationale': rationale}
-            set_values.append(set_value)
+        # for versions prior to 0.1.59 include parameters
+        if self._check_version < 1059:
+            for set_parameter in self._profile.modify.set_parameters:
+                name = self._format_osco_rule_name(set_parameter.param_id)
+                parameter_value = set_parameter.values[0]
+                value = parameter_value.__root__
+                rationale = self._get_rationale_for_set_value()
+                set_value = {'name': name, 'value': value, 'rationale': rationale}
+                set_values.append(set_value)
         return set_values
 
-    def _get_metadata_prop_value(self, profile, name, default_) -> str:
+    def _format_osco_rule_name(self, name: str) -> str:
+        """Format for OSCO.
+
+        1. remove prefix xccdf_org.ssgproject.content_rule_
+        2. change underscores to dashes
+        3. add prefix ocp4-
+        """
+        return 'ocp4-' + (name.replace('xccdf_org.ssgproject.content_rule_', '').replace('_', '-'))
+
+    def _get_metadata_prop_value(self, name: str, default_: str) -> str:
         """Extract metadata prop or else default if not present."""
-        if profile.metadata.props is not None:
-            for prop in profile.metadata.props:
+        if self._profile.metadata.props is not None:
+            for prop in self._profile.metadata.props:
                 if prop.name == name:
                     return prop.value
         logger.info(f'using default: {name} = {default_}')
         return default_
 
-    def _get_disable_rules(self, profile) -> List[str]:
+    def _get_disable_rules(self) -> List[str]:
         """Extract disabled rules."""
         value = []
-        if profile.imports is not None:
-            for item in profile.imports:
+        if self._profile.imports is not None:
+            for item in self._profile.imports:
                 if item.exclude_controls is not None:
                     for control in item.exclude_controls:
-                        if control.with_ids is not None:
-                            for with_id in control.with_ids:
-                                name = with_id.__root__
-                                rationale = self._get_rationale_for_disable_rule()
-                                entry = {'name': name, 'rationale': rationale}
-                                value.append(entry)
+                        self._add_disable_rules_for_control(value, control)
         return value
 
+    def _add_disable_rules_for_control(self, value, control):
+        """Extract disabled rules for control."""
+        if control.with_ids is not None:
+            for with_id in control.with_ids:
+                name = self._format_osco_rule_name(with_id.__root__)
```

Comment:
A slightly cleaner way of doing these things is with a function I added, as_list.  as_list(foo) will return [] if foo is None - otherwise it returns foo.  This removes the outer if None check and makes things more declarative.  So you can just say `for with_id in as_list(control.with_ids)`

Reply:
Nice! Fixed.